packages/amplify-migration-template-gen/src/cfn-stack-refactor-updater.ts

```diff
@@ -78,6 +79,16 @@ export async function tryRefactorStack(
       },
     ];
   }
+
+  const sourceStackName = createStackRefactorCommandInput.StackDefinitions?.[0].StackName;
```

Comment:
this ensures refactor waits for stack to get into update complete state to avoid subsequent refactors to run into stack update in progress errors.